Libraries/LibJS/Runtime/StringPrototype.cpp

```diff
@@ -507,6 +520,88 @@ JS_DEFINE_NATIVE_FUNCTION(StringPrototype::slice)
     return js_string(vm, string_part);
 }
 
+JS_DEFINE_NATIVE_FUNCTION(StringPrototype::split)
+{
+    auto string = ak_string_from(vm, global_object);
+    if (string.is_null())
+        return {};
+
+    auto* result = Array::create(global_object);
+    size_t result_len = 0;
+
+    size_t limit = NumericLimits<size_t>::max();
+    if (!vm.argument(1).is_undefined())
+        limit = vm.argument(1).to_size_t(global_object);
```

Comment:
`to_size_t` is non-standard and being phased out, please don't use it in new code. The spec says this should use the [`ToUint32()` abstract operation](https://tc39.es/ecma262/#sec-touint32), so let's add `Value::to_u32()` and use that instead. The implementation can be similar to `to_size_t`'s, but there are subtle differences - this would handle infinity incorrectly, for example.